easybuild/tools/easystack.py

```
@@ -0,0 +1,186 @@
+
+                eb = Easystack()
+        except Exception:
+            raise EasyBuildError("Could not read provided easystack.")
+
+        sw_dict = spec_dict["software"]
+
+        # assign software-specific EB attributes
+        for software in sw_dict:
+            asterisk_err = "Easystack specifications of '%s' contain asterisk. " % (str(software))
+            asterisk_err += "Wildcard feature is not supported yet."
+            wrong_structure_err = "Easystack specifications of '%s' have wrong yaml structure." % (str(software))
+            try:
+                # iterates through toolchains to find out what sw version is needed
+                for yaml_tc in sw_dict[software]['toolchains']:
+                    # if version string containts asterisk or labels, raise error (not implemented yet)
+                    if '*' in str(sw_dict[software]['toolchains'][yaml_tc]['versions']):
+                        raise EasyBuildError(asterisk_err)
+                    for yaml_version in sw_dict[software]['toolchains'][yaml_tc]['versions']:
+                        try:
+                            yaml_ver_specs = sw_dict[software]['toolchains'][yaml_tc]['versions'][yaml_version]
+                            if 'versionsuffix' in str(yaml_ver_specs):
+                                ver_suf_err = "Easystack specifications of '%s' contain versionsuffix. " % str(software)
+                                ver_suf_err += "This isn't supported yet."
+                                raise EasyBuildError(ver_suf_err)
+                            elif 'exclude-labels' in str(yaml_ver_specs) or 'include-labels' in str(yaml_ver_specs):
+                                lab_err = "Easystack specifications of '%s' contain labels. " % str(software)
+                                lab_err += "Labels aren't supported yet."
+                                raise EasyBuildError(lab_err)
+                        except TypeError:
```

Again, please don't do this. If there's a `TypeError`, something is very wrong, and you should silently ignore it?!

---

please check if it is ok in the next iteration